overlay/generic/smartdc/lib/smartos_prompt_config.sh

```diff
@@ -30,73 +47,137 @@ dns_resolver2="8.8.4.4"
 declare -a states
 declare -a nics
 declare -a assigned
-declare -a DISK_LIST
+declare prmpt_str
+
+nicsup_done=0
 
-sigexit()
+fatal()
 {
 	echo
-	echo "System configuration has not been completed."
+	if [[ -n "$1" ]]; then
+		echo "ERROR: $1"
+	fi
+	echo
+	echo "System configuration failed, launching a shell."
 	echo "You must reboot to re-run system configuration."
-	exit 0
+	echo "A log of setup may be found in /var/log/prompt-config.log"
+	echo
+	/usr/bin/bash
 }
 
-#
-# Get the max. IP addr for the given field, based in the netmask.
-# That is, if netmask is 255, then its just the input field, otherwise its
-# the host portion of the netmask (e.g. netmask 224 -> 31).
-# Param 1 is the field and param 2 the mask for that field.
-#
-max_fld()
+
+sig_doshell()
 {
-	if [ $2 -eq 255 ]; then
-		fmax=$1
-	else
-		fmax=$((255 & ~$2))
-	fi
+	echo
+	echo
+	echo "Bringing up a shell.  When you are done in the shell hit ^D to"
+	echo "return to the system configuration tool."
+	echo
+
+	/usr/bin/bash
+
+	echo
+	echo "Resuming the system configuration tool."
+	echo
+	printf "$prmpt_str"
```

Comment:
`printf` will interpret special strings like `%s`, `%c`, etc. inside `$prmpt_str`.  you should use

``` bash
printf '%s' "$prmpt_str"
```